tests/benchmark/bm_basics.h

```diff
@@ -0,0 +1,130 @@
+        VecSimQueryResult_Free(hnsw_results);
+        iter++;
+        st.ResumeTiming();
+    }
+    st.counters["Avg. results number"] = (double)total_res / iter;
+    st.counters["Recall"] = (float)total_res / total_res_bf;
+}
+
+#define UNIT_AND_ITERATIONS                                                                        \
+    Unit(benchmark::kMillisecond)->Iterations((long)BM_VecSimGeneral::block_size)
+
+// The actual radius will be the given arg divided by 100, since arg must be an integer.
+#define REGISTER_Range_BF(BM_FUNC)                                                                 \
+    BENCHMARK_REGISTER_F(BM_VecSimBasics, BM_FUNC)                                                 \
+        ->Arg(20)                                                                                  \
+        ->ArgName("radiusX100")                                                                    \
+        ->Arg(35)                                                                                  \
+        ->ArgName("radiusX100")                                                                    \
+        ->Arg(50)                                                                                  \
+        ->ArgName("radiusX100")                                                                    \
+        ->Unit(benchmark::kMillisecond)
+
+#define HNSW_RANGE_ARGS(radius, epsilon)                                                           \
+    Args({radius, epsilon})->ArgNames({"radiusX100", "epsilonX1000"})
+
+// {radius*100, epsilon*1000}
+// The actual radius will be the given arg divided by 100, and the actual epsilon values
+// will be the given arg divided by 1000.
+#define REGISTER_Range_HNSW(BM_FUNC)                                                               \
+    BENCHMARK_REGISTER_F(BM_VecSimBasics, BM_FUNC)                                                 \
+        ->HNSW_RANGE_ARGS(20, 1)                                                                   \
+        ->HNSW_RANGE_ARGS(20, 10)                                                                  \
+        ->HNSW_RANGE_ARGS(20, 100)                                                                 \
+        ->HNSW_RANGE_ARGS(35, 1)                                                                   \
+        ->HNSW_RANGE_ARGS(35, 10)                                                                  \
+        ->HNSW_RANGE_ARGS(35, 100)                                                                 \
+        ->HNSW_RANGE_ARGS(50, 1)                                                                   \
+        ->HNSW_RANGE_ARGS(50, 10)                                                                  \
+        ->HNSW_RANGE_ARGS(50, 100)                                                                 \
+        ->Iterations(100)                                                                          \
+        ->Unit(benchmark::kMillisecond)
```

Comment:
```suggestion
    BENCHMARK_REGISTER_F(BM_VecSimBasics, BM_FUNC)                                                 \
        ->ArgPair(20, 1)                                                                   \
        ->ArgPair(20, 10)                                                                  \
        ->ArgPair(20, 100)                                                                 \
        ->ArgPair(35, 1)                                                                   \
        ->ArgPair(35, 10)                                                                  \
        ->ArgPair(35, 100)                                                                 \
        ->ArgPair(50, 1)                                                                   \
        ->ArgPair(50, 10)                                                                  \
        ->ArgPair(50, 100)                                                                 \
        ->Iterations(100)                                                                          \
        ->ArgNames({"radiusX100", "epsilonX1000"})                            \
        ->Unit(benchmark::kMillisecond)
```

Reply:
from benchmark.h: 

   // Equivalent to Args({x, y})
  // NOTE: This is a legacy C++03 interface provided for compatibility only.
  //   New code should use 'Args'.